addons/block_code/drag_manager/drag_manager.gd

```diff
@@ -9,111 +9,197 @@ signal block_modified
 @export var block_canvas_path: NodePath
 
 const Constants = preload("res://addons/block_code/ui/constants.gd")
-const BLOCK_AUTO_PLACE_MARGIN: Vector2 = Vector2(16, 8)
 
-var drag_offset: Vector2
-var dragging: Block = null
+enum DragAction { NONE, PLACE, REMOVE }
+
+
+class Drag:
+	extends Control
+	var _block: Block
+	var _block_canvas: BlockCanvas
+	var _preview_block: Control
+	var action: DragAction:
+		get:
+			return action
+		set(value):
+			if action != value:
+				action = value
+				_update_action_hint()
+
+	var target_snap_point: SnapPoint:
+		get:
+			return target_snap_point
+		set(value):
+			if target_snap_point != value:
+				target_snap_point = value
+				_update_preview()
+
+	var snap_block: Block:
+		get:
+			return target_snap_point.block if target_snap_point else null
+
+	func _init(block: Block, offset: Vector2, block_canvas: BlockCanvas):
+		assert(block.get_parent() == null)
+
+		add_child(block)
+		block.position = -offset
+
+		_block = block
+		_block_canvas = block_canvas
+
+	func apply_drag() -> Block:
+		if action == DragAction.PLACE:
+			_place_block()
+			return _block
+		elif action == DragAction.REMOVE:
+			_remove_block()
+			return null
+		else:
+			return null
+
+	func _remove_block():
+		target_snap_point = null
+		_block.g()
```

Comment:
When the block becomes removed, it shows error:
`res://addons/block_code/drag_manager/drag_manager.gd:62 - Invalid call. Nonexistent function 'g' in base 'MarginContainer (EntryBlock)'.`

Reply:
Oh that's so weird! What did I do to that commit? :/ Thanks for spotting. I fixed that in the newest commit :)